fuel-p2p/src/gossipsub/config.rs

```diff
@@ -51,15 +53,42 @@ pub(crate) fn default_gossipsub_config() -> GossipsubConfig {
 
 /// Given a `P2pConfig` creates a Gossipsub Behaviour
 pub(crate) fn build_gossipsub_behaviour(p2p_config: &P2PConfig) -> Gossipsub {
-    let mut gossipsub = Gossipsub::new(
-        MessageAuthenticity::Signed(p2p_config.local_keypair.clone()),
-        p2p_config.gossipsub_config.clone(),
-    )
-    .expect("gossipsub initialized");
+    if p2p_config.metrics {
+        // Move to Metrics related feature flag
+        let mut p2p_registry = Registry::default();
+
+        let metrics_config = MetricsConfig::default();
+
+        let mut gossipsub = Gossipsub::new_with_metrics(
+            MessageAuthenticity::Signed(p2p_config.local_keypair.clone()),
+            p2p_config.gossipsub_config.clone(),
+            &mut p2p_registry,
+            metrics_config,
+        )
+        .expect("gossipsub initialized");
+
+        // This couldn't be set unless multiple p2p services are running? So it's ok to unwrap
+        P2P_METRICS
+            .gossip_sub_registry
+            .set(Box::new(p2p_registry))
+            .unwrap_or(());
+
+        gossipsub
+            .with_peer_score(PeerScoreParams::default(), PeerScoreThresholds::default())
+            .expect("gossipsub initialized with peer score");
+
+        gossipsub
+    } else {
+        let mut gossipsub = Gossipsub::new(
+            MessageAuthenticity::Signed(p2p_config.local_keypair.clone()),
+            p2p_config.gossipsub_config.clone(),
+        )
+        .expect("gossipsub initialized");
```

Comment:
Is panicking here what we want?

Reply:
[This `.expect` is mirrored on the non-metrics version](https://github.com/FuelLabs/fuel-core/blob/master/fuel-p2p/src/gossipsub/config.rs#L58), so I'm carrying out over into initialization with metrics. However, since it happens in the existing non-metrics initialization I wouldn't expect the added one to introduce any additional problems.